### DISCUSSION TITLE
What do you use AccessibilitySnapshot for?

### DISCUSSION THREAD

**Discussion by NickEntin** [cashapp/AccessibilitySnapshot#103] [General]
If you use AccessibilitySnapshot in your projects, I'd love to know what app(s) you use it for and which features of the framework you're using. This helps shape the way we think about things like what iOS versions we support, which features we prioritize, etc.

And if you aren't currently using AccessibilitySnapshot, what's keeping you from adopting it in your projects? Are there specific issues we can address to help make adoption easier?

Thanks for helping make AccessibilitySnapshot better!

**Comment by Sherlouk** [comment]
In my own projects I use the project in a few different ways. 

One of the most common ways is I actually use the underlying `AccessibilitySnapshotView`, wrap it in UIViewRepresentable, and embed it in my SwiftUI previews. This enables me to visualise my views with a VoiceOver lens updating in real-time. This makes it very easy to spot mistakes.

Of course I do also use the SnapshotTesting strategy too, generating snapshots of my key journeys and allowing me to quickly spot visual and accessibility regressions. I also use the underlying classes to create a textual representation for where the order of elements and data hierarchy is important.

Really I suppose it's about visualising what is an otherwise very difficult part of the UI to test. By visualising it, it highlights mistakes a lot easier (such as not adding labels to images, etc).

It might be nice if there was some ability to assert/fail if certain criteria isn't met - but I can see how this leaks beyond the pure "accessibility snapshot" theme.

**Comment by ZevEisenberg** [comment]
NYT Games and Crossplay are both using it for new stuff, especially SwiftUI views! We started using it when we did the Games app home screen redesign a few years back.
